app/helpers/markup_helper.rb

```diff
@@ -1,10 +1,13 @@
 module MarkupHelper
   def editor_paths
+    preview_params = {}
+    preview_params[:project_id] = params[:project_id] if params[:project_id]
+
     {
-      form_url: form_fields_path(format: :js),
-      help_url: help_markup_path,
-      preview_url: preview_markup_path,
-      source_url: source_fields_path
+      form_url: main_app.form_fields_path(format: :js),
+      help_url: main_app.help_markup_path,
+      preview_url: main_app.preview_markup_path(preview_params),
```

Comment:
What was the catalyst for this change. We're rendering somewhere outside of a project with the full editor?

Reply:
Most Tylium controllers are aware of the project context via ProjectScoped, but MarkupController that renders the Preview isn't. We need for a way to let Preview know about the project context we're in.